src/core/server/csp/csp_config.ts

```diff
@@ -66,12 +67,19 @@ export class CspConfig implements ICspConfig {
    * Returns the default CSP configuration when passed with no config
    * @internal
    */
-  constructor(rawCspConfig: Partial<Omit<ICspConfig, 'header'>> = {}) {
+  constructor(rawCspConfig: Partial<Omit<ICspConfig, 'header'>> = {}, env?: Env) {
```

Comment:
I know this means editing a few more test files, but `env` should probably not be optional here. Else there is a risk to instantiate without it from somewhere and uses wrong csp rules.

Reply:
There is no valid case when it can be optional since `env` is always defined in coreContext. https://github.com/elastic/kibana/pull/54991/files#diff-748376502c51ed8bfc271f4a6c4c26d3R89

Reply:
Except for the static `DEFAULT` property, which seemed necessary. That instance of `CspConfig` can't be created if the env is required.

Reply:
Took another look at the uses of `CspConfig.DEFAULT` and it doesn't look like the platform actually requires that field, so I can probably get away with removing it, will give it a shot.

Reply:
Hum, good point. Except I *think* we are going to need the `DEFAULT` to also be aware of `Env`, as it can change the rules and therefor the header

https://github.com/elastic/kibana/blob/614bde927efc34c9d90b0984da3e1f2a51a784b1/src/legacy/core_plugins/kibana/server/lib/csp_usage_collector/csp_collector.ts#L34-L38

If I understand correctly, in distribution, the usage collector will always have `rulesChangedFromDefault` being true, as `CspConfig.DEFAULT` is not aware of whether we are in distribution mode or not...

@restrry @joshdover : @spalger was saying this was quite urgent, do you have any idea on solution? should http service expose a `defaultCsp` in addition to the actual `csp` it's exposing atm. to be able to have a non-static default that could be instanciated with `Env`?

Reply:
Misread the condition, it's actually in non-distribution that the issue would occur. I may be less important?

Reply:
Yeah, I aimed to keep the default the same as the distribution. My plan is to expose the `rulesChangedFromDefault` state from the csp service directly rather than exposing access to the default via `CspConfig.DEFAULT`, working on it now.

Reply:
Alright, I've removed the `CspConfig.DEFAULT` static property and instead expose the `headerChangedFromDefault` property on CspConfig itself. Also included a `env.mock.ts` with `createMockEnv()` to provide the env from tests.